src/FromData.purs

```diff
@@ -2,190 +2,253 @@ module FromData
   ( FromDataError(..)
   , class FromData
   , class FromDataArgs
+
   , class FromDataArgsRL
-  , class FromDataWithIndex
+  , class FromDataWithSchema
   , fromData
   , fromDataArgs
   , fromDataArgsRec
-  , fromDataWithIndex
+  , fromDataWithSchema
   , genericFromData
   ) where
 
-import Prelude
-
-import ConstrIndices (class HasConstrIndices, constrIndices)
-import Data.Show.Generic (genericShow)
-import Control.Alternative ((<|>), guard)
+import Control.Alternative ((<|>))
 import Data.Array (uncons)
 import Data.Array as Array
-import Data.NonEmpty (NonEmpty(NonEmpty))
 import Data.BigInt (BigInt)
 import Data.BigInt as BigInt
 import Data.Either (Either(Left, Right), hush, note)
 import Data.Generic.Rep as G
 import Data.List (List)
-import Data.Map as Map
 import Data.Maybe (Maybe(Nothing, Just), maybe)
 import Data.Newtype (unwrap)
+import Data.NonEmpty (NonEmpty(NonEmpty))
 import Data.Ratio (Ratio, reduce)
+import Data.Show.Generic (genericShow)
 import Data.Symbol (class IsSymbol, reflectSymbol)
 import Data.TextDecoder (decodeUtf8)
 import Data.Traversable (traverse)
 import Data.Tuple (Tuple(Tuple))
 import Data.UInt (UInt)
 import Data.Unfoldable (class Unfoldable)
 import Helpers (bigIntToUInt)
+import Prelude
```

Comment:
Normally I'd encourage explicit imports, but for `Prelude` I don't think it's necessary (and makes it harder to keep track of redundant or missing imports)